File: Homework4/average.cpp
#include<iostream>
#include<vector>
#include<string>
#include<fstream>
#include<sstream>
#include<stdio.h>

class StudentRecord{
     public:
     std::string first_name, last_name;
     float score;
};

float calculate_average(std::vector<StudentRecord> *rost){
     float sum=0;
     int NoStds=(*rost).size();
     for(int i=0; i<NoStds; i++){
          sum+=(*rost)[i].score;
     }
     return sum/NoStds;
}

int main(int argc, char* argv[]){
     std::vector<StudentRecord> roster;
     std::ifstream infile(argv[1]);
     std::string ln;
     while(std::getline(infile, ln)){
          StudentRecord rec;
          std::string sval;
          std::istringstream stringstream(ln);
          std::getline(stringstream, sval, ',');
          rec.last_name=sval;
          std::getline(stringstream, sval, ',');
          rec.first_name=sval;
          std::getline(stringstream, sval, ',');
          rec.score=std::stof(sval);
          roster.push_back(rec);
     }
     float avg=calculate_average(&roster);
     printf("Student Average: %f%%\n",avg);
}

File: Homework5/average.py
#!/usr/bin/python

import sys

class StudentRecord:
     def __init__(self, last_name, first_name, score):
          self.first_name=first_name
          self.last_name=last_name
          self.score=score

def calculate_average(roster):
     sum=0
     for sdnt in roster:
          sum+=sdnt.score
     return sum/len(roster)

def main(fname):
     roster = []
     lns=open(fname).readlines()
     for sdnt in lns:
          sval=sdnt.split(",")
          roster.append(StudentRecord(sval[0],sval[1],float(sval[2])))
     print("Class Average: %f%%" % calculate_average(roster))

if __name__=="__main__":
     main(sys.argv[1])
